Implement text attributes in displayPort_t functions

fiam opened the pull request:
Allows using hardware based blink, character color inversion and solid backgrounds.

fiam committed:
Add textAttributes_t, update display port system and drivers for it

- Add textAttributes_t, implemented as a single byte which holds
attributes for blink, character color inversion and solig background.
- Update display port drivers to accept atextAttributes_t in both
writeString() and writeChar(). Currently all drivers accept but
ignore this new parameter.
- Add displayWriteAttr() and displayWriteCharAttr() functions, which
accept a textAttributes_t argument. Functions without the attribute
parameter just call the display port driver writing functions with
 TEXT_ATTRIBUTES_NONE as the attrs parameter.

fiam committed:
Add support for character attributes to MAX7456 drivers

Support blink, inverse and solid bg character attributes. No
code using them yet.

fiam committed:
Implement OSD blink using hardware blink support

Use textAttributes_t to specify which fields should blink. This
gets rid of osdResetAlarms() and osdUpdateAlarms() since now
each OSD element handles its own blinking.

Fixes #2088

digitalentity commented on src/main/drivers/display.c:
```diff
@@ -77,14 +77,28 @@ int displayWrite(displayPort_t *instance, uint8_t x, uint8_t y, const char *s)
+    return instance->vTable->writeString(instance, x, y, s, TEXT_ATTRIBUTES_NONE);
+}
+
+int displayWriteAttr(displayPort_t *instance, uint8_t x, uint8_t y, const char *s, textAttributes_t attr)
```
`displayWriteAttr` name is confusing - it gives an impression that function will only write attributes. Probably `displayWriteWithAttr` is a better name?

fiam replied:
Seems like a good idea. I've just added a commit renaming those functions.

digitalentity commented on src/main/drivers/display.c:
```diff
@@ -77,14 +77,28 @@ int displayWrite(displayPort_t *instance, uint8_t x, uint8_t y, const char *s)
+    return instance->vTable->writeChar(instance, x, y, c, TEXT_ATTRIBUTES_NONE);
+}
+
+int displayWriteCharAttr(displayPort_t *instance, uint8_t x, uint8_t y, uint8_t c, textAttributes_t attr)
```
Same as above

digitalentity commented on src/main/drivers/max7456.c:
```diff
@@ -183,12 +188,12 @@ uint16_t maxScreenSize = VIDEO_BUFFER_CHARS_PAL;
 // in screenIsDirty to upgrade only changed chars this solution
 // is faster than redrawing the whole screen on each frame
-static uint8_t screenBuffer[VIDEO_BUFFER_CHARS_PAL] ALIGNED(4);
+static uint16_t screenBuffer[VIDEO_BUFFER_CHARS_PAL] ALIGNED(4);
```
As we discussed on Slack. Probably it would be better to store attributes in a separate buffer and pack attributes to two adjacent characters into a single `uint8_t`.

fiam replied:
As discussed on Slack a bit later, we end up using a lot of FLASH to save those 240 bytes of BSS, so it seems not worth it right now. Plus, we had already saved ~600 bytes from BSS when moving from the double buffer to a bitarray, so we still reduced the memory used by the MAX7456 overall.

fiam committed:
Rename displayWriteAttr() to displayWriteWithAttr()

Also, displayWriteCharAttr() to displayWriteCharWithAttr().
Otherwise, the names might be interpreted as writing just the
attributes, while they actually write the text AND its attributes.

fiam committed:
Remove UNUSED() for arguments that are now actually used

Added in the 1st commit, which just added the new functions but
didn't include the implementation to use the textAttributes_t
arguments in the MAX7456 driver.
